pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -578,20 +579,28 @@ def __resolve_ckpt_dir(self, trainer: "pl.Trainer") -> None:
         """
         if self.dirpath is not None:
             return  # short circuit
-
-        if trainer.logger is not None:
+        if trainer.loggers:
             if trainer.weights_save_path != trainer.default_root_dir:
                 # the user has changed weights_save_path, it overrides anything
                 save_dir = trainer.weights_save_path
-            else:
+            elif len(trainer.loggers) == 1:
                 save_dir = trainer.logger.save_dir or trainer.default_root_dir
+            else:
+                save_dir = trainer.default_root_dir
 
+            name = (
+                trainer.loggers[0].name
+                if len(trainer.loggers) == 1
+                else "_".join(dict.fromkeys(str(logger.name) for logger in trainer.loggers))
+            )
             version = (
-                trainer.logger.version
-                if isinstance(trainer.logger.version, str)
-                else f"version_{trainer.logger.version}"
+                trainer.loggers[0].version
+                if len(trainer.loggers) == 1
+                else "_".join(dict.fromkeys(str(logger.version) for logger in trainer.loggers))
```

Comment:
```suggestion
                "_".join(str(logger.version) for logger in trainer.loggers)
```
is this different?

Reply:
`dict.fromkeys` was added to prevent duplicate names and versions. You can see the original PR here https://github.com/PyTorchLightning/pytorch-lightning/pull/10976. Since this is the original behavior in `LoggerCollection`, I think we should leave this as is to keep it a pure refactor.

Reply:
how about `set` then? just trying to avoid creating a dict since we are only concerned with keys. works for both single and multiple loggers.

Reply:
I tested it out and this actually results in a different behavior. It looks like `dict.fromkeys()` will remove duplicates and preserve the order, while `set` only removes duplicates. This is because dictionaries are ordered as of Python 3.7.

Reply:
Oh! Yeah that's true. Totally forgot about it.

Reply:
Yeah this behavior is not very transparent so I added a comment to explain it